include/libtorrent/socket.hpp

```diff
@@ -63,6 +63,13 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "simulator/simulator.hpp"
 #endif
 
+#if TORRENT_USE_NETLINK
+#include <linux/netlink.h>
+#ifndef SOL_NETLINK
+#define SOL_NETLINK 270
```

Comment:
this looks suspicious. I take it this is android specific, is it?

Reply:
agree, actually, the problem is in linux, I will try adding more headers

Reply:
I tried multiple variants/includes in my travis account, and still fails, it looks like I'm not the only one. @ssiloti what do you think?

Reply:
`sys/socket.h` is the include you want to get the definition of `SOL_NETLINK`. This is with glibc though, Android may be different.

Reply:
I tried that already, but I will try again, just to make sure

Reply:
yes, my latest travis job has that include, and it fails, what other option do I have?

Reply:
Looks like glibc is not consistent with defining `SOL_NETLINK` and we're not the only ones resorting to defining it ourselves. It's unfortunate, but seems to be necessary.

Reply:
is netlink a linux specific API? is it safe to assume it uses a stable value?

Reply:
I don't think it's linux specific, but more to linux kernel based OSes, but it's not mandatory. I think the value is pretty stable, I mean, fixed and copied everywhere.

Reply:
Yes, netlink is a Linux specific API. It is safe to assume the value is stable given Linux's strong commitment to maintaining a stable userspace ABI.